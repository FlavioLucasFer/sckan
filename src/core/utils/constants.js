const MAIN_COLOR = '#1ac3d6';
const SECUNDARY_COLOR = '#245257';

export {
    MAIN_COLOR,
    SECUNDARY_COLOR,
};