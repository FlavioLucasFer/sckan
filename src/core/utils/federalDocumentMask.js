'use strict';

function federalDocumentMask(value) {
	value = value.replace(/\D/g, '');

	if (value.length <= 11) { //CPF
		value = value.replace(/(\d{3})(\d)/, '$1.$2')
			.replace(/(\d{3})(\d)/, '$1.$2')
			.replace(/(\d{3})(\d{1,2})$/, '$1-$2');
	} else { //CNPJ
		value = value.replace(/^(\d{2})(\d)/, '$1.$2')
			.replace(/^(\d{2})\.(\d{3})(\d)/, '$1.$2.$3')
			.replace(/\.(\d{3})(\d)/, '.$1/$2')
			.replace(/(\d{4})(\d)/, '$1-$2')
			.replace(/(-\d{2})\d+?$/, '$1');
	}

	return value;
}

export default federalDocumentMask;
