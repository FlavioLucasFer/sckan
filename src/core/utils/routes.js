const HOME_ROUTE = '/';
const LOGIN_ROUTE = '/login';
const COMPANY_ROUTE = '/companies';
const COMPANY_FORM_ROUTE = `${COMPANY_ROUTE}/form`;
const PROJECT_ROUTE = '/projects';
const PROJECT_FORM_ROUTE = `${PROJECT_ROUTE}/form`;
const SPRINT_ROUTE = '/sprints';
const SPRINT_FORM_ROUTE = `${SPRINT_ROUTE}/form`;
const TASK_ROUTE = '/tasks';
const TASK_FORM_ROUTE = `${TASK_ROUTE}/form`;
const USER_ROUTE = '/users';
const USER_FORM_ROUTE = `${USER_ROUTE}/form`;
const USER_PROFILE = '/user';
const USER_SITTINGS = `${USER_PROFILE}/settings`;

export {
	HOME_ROUTE,
  LOGIN_ROUTE,
  COMPANY_ROUTE,
	COMPANY_FORM_ROUTE,
  PROJECT_ROUTE,
	PROJECT_FORM_ROUTE,
  SPRINT_ROUTE,
	SPRINT_FORM_ROUTE,
  TASK_ROUTE,
	TASK_FORM_ROUTE,
  USER_ROUTE,
	USER_FORM_ROUTE,
	USER_PROFILE,
	USER_SITTINGS,
};