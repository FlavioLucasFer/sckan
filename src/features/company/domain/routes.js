'use strict';

const { REACT_APP_API_URL } = process.env;

const BASE_URL = `${REACT_APP_API_URL}/companies`;

const ROUTES = {
	CREATE: `${BASE_URL}/create`,
	EDIT: `${BASE_URL}/edit`,
	FIND: `${BASE_URL}/find`,
	FIND_BY_ID: `${BASE_URL}/find/:id`,
	REMOVE: `${BASE_URL}/remove`,
};

export default ROUTES;
