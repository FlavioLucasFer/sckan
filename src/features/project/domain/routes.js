'use strict'

const { REACT_APP_API_URL } = process.env;

const BASE_URL = `${REACT_APP_API_URL}/projects`;

const ROUTES = {
	CREATE: `${BASE_URL}/create`,
	EDIT: `${BASE_URL}/edit`,
	REMOVE: `${BASE_URL}/remove`,
	FIND: `${BASE_URL}/find`,
};

export default ROUTES;
