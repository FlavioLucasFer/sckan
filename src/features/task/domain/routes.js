'use strict'

const { REACT_APP_API_URL } = process.env;

const BASE_URL = `${REACT_APP_API_URL}/tasks`;

const ROUTES = {
	CREATE: `${BASE_URL}/create`,
	EDIT: `${BASE_URL}/edit`,
	REMOVE: `${BASE_URL}/remove`,
	FIND: `${BASE_URL}/find`,
	FIND_BY_ID: `${BASE_URL}/find/:id`,
};

export default ROUTES;
