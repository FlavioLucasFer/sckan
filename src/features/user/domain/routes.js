'use strict'

const { REACT_APP_API_URL } = process.env;

const BASE_URL = `${REACT_APP_API_URL}/users`;

const ROUTES = {
	CREATE: `${BASE_URL}/create`,
	EDIT: `${BASE_URL}/edit`,
	REMOVE: `${BASE_URL}/remove`,
	FIND: `${BASE_URL}/find`,
	FIND_BY_ID: `${BASE_URL}/find/:id`,
	SIGN_IN: `${BASE_URL}/sign-in`,
	LOGOUT: `${BASE_URL}/logout`,
};

export default ROUTES;
