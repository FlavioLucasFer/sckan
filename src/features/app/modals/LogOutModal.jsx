import React from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { useHistory } from 'react-router-dom';

import ConfirmActionModal from 'core/modals/ConfirmActionModal';
import { LOGIN_ROUTE } from 'core/utils/routes';

import userService from 'features/user/domain/service';

import { setCurrentScreen } from 'redux/reducers/currentScreen';

LogOutModal.propTypes = {
	id: PropTypes.string.isRequired,
};

function LogOutModal({ id }) {
	const dispatch = useDispatch();
	const { push } = useHistory();

	async function handleConfirm() {
		try {
			await userService.logout();
		} catch (err) {
			console.log('err:', err);
		}

		push(LOGIN_ROUTE);
		dispatch(setCurrentScreen(LOGIN_ROUTE));
	}

	return (
		<ConfirmActionModal id={id}
			title="Are you sure you want logout?" 
			onConfirm={handleConfirm} />
	);
}

export default LogOutModal;