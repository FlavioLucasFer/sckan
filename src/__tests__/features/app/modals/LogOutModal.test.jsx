import React from 'react';
import { shallow } from 'enzyme';

import LogOutModal from 'features/app/modals/LogOutModal';
import { HOME_ROUTE, LOGIN_ROUTE } from 'core/utils/routes';

const mockHistory = {
	push: jest.fn(),
};

const mockLocation = {
	pathname: HOME_ROUTE,
}

jest.mock('react-router-dom', () => ({
	...(jest.requireActual('react-router-dom')),
	useHistory: () => (mockHistory),
	useLocation: () => (mockLocation),
}));

describe('Testing LogOutModal app feature modal component', () => {
	afterAll(() => {
		jest.clearAllMocks();
	});
	
	it('should render correctly', () => {
		const wrapper = shallow(
			<LogOutModal id="logout-modal-test" />
		);

		expect(wrapper.debug()).toMatchSnapshot();
	});

	it('should navigate to login screen when confirm action', () => {
		const wrapper = shallow(
			<LogOutModal id="logout-modal-test" />
		);

		wrapper.find({ id: 'logout-modal-test' })
			.dive()
			.find({ id: 'logout-modal-test' })
			.dive()
			.find({ title: 'Yes' })
			.simulate('click');

		expect(mockHistory.push).toBeCalledWith(LOGIN_ROUTE);
	});

	it('should stay in current screen when cancel action', () => {
		const wrapper = shallow(
			<LogOutModal id="logout-modal-test" />
		);

		wrapper.find({ id: 'logout-modal-test' })
			.dive()
			.find({ id: 'logout-modal-test' })
			.dive()
			.find({ title: 'No' })
			.simulate('click');

		expect(mockLocation.pathname).toBe(HOME_ROUTE);
	});
});
