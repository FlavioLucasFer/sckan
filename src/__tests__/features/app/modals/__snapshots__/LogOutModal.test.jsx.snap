// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing LogOutModal app feature modal component should render correctly 1`] = `"<ConfirmActionModal id=\\"logout-modal-test\\" title=\\"Are you sure you want logout?\\" onConfirm={[Function: handleConfirm]} />"`;
